packages/editor/src/editor/PortableTextEditor.tsx

```diff
@@ -1,460 +1,314 @@
+     */
+    useImperativeHandle(forwardedRef, () => instance, [instance])
 
     return (
       <>
-        {legacyPatches ? (
-          <RoutePatchesObservableToEditorActor
-            editorActor={this.editorActor}
-            patches$={legacyPatches}
-          />
-        ) : null}
-        <EditorActorContext.Provider value={this.editorActor}>
+        <EditorActorContext.Provider value={instance.editorActor}>
           <Slate
-            editor={this.slateEditor.instance}
-            initialValue={this.slateEditor.initialValue}
+            editor={instance.slateEditor.instance}
+            initialValue={instance.slateEditor.initialValue}
           >
-            <PortableTextEditorContext.Provider value={this}>
+            <PortableTextEditorContext.Provider value={instance}>
               <PortableTextEditorSelectionProvider
-                editorActor={this.editorActor}
+                editorActor={instance.editorActor}
               >
                 <Synchronizer
-                  editorActor={this.editorActor}
-                  getValue={this.getValue}
+                  editorActor={instance.editorActor}
+                  getValue={instance.getValue}
                   onChange={(change) => {
-                    if (!this.props.editor) {
-                      this.props.onChange(change)
-                    }
                     /**
                      * For backwards compatibility, we relay all changes to the
                      * `change$` Subject as well.
                      */
-                    this.change$.next(change)
+                    instance.change$.next(change)
                   }}
-                  value={this.props.value}
+                  value={props.value}
                 />
-                {this.props.children}
+                {props.children}
               </PortableTextEditorSelectionProvider>
             </PortableTextEditorContext.Provider>
           </Slate>
         </EditorActorContext.Provider>
       </>
     )
-  }
+  },
+)
+PortableTextEditorWithEditorProp.displayName =
+  'ForwardRef(PortableTextEditorWithEditorProp)'
+
+const PortableTextEditorComponent = forwardRef(
+  (
+    props: PortableTextEditorProps<Editor | undefined>,
+    forwardedRef: ForwardedRef<PortableTextEditorInstance>,
+  ) => {
+    const ref = useRef<PortableTextEditorInstance | null>(null)
+
+    /**
+     * Forward ref to the instance, this mimics the previous behaviour that allowed consumers to get a class instance by setting a ref
+     */
+    useImperativeHandle(forwardedRef, () => ref.current!)
 
-  // Static API methods
-  static activeAnnotations = (
-    editor: PortableTextEditor,
+    /**
+     * Unclear if `editorRef` is really needed, as setting a regular `ref` has the same effect (and always has been)
+     */
+    useImperativeHandle(props.editorRef, () => ref.current!)
+
```

Comment:
I don't think we need `editorRef` @christianhg, as we can achieve the same thing with just `ref` 🙌